examples/experimental/i3s-picking/app.js

```diff
@@ -196,19 +196,43 @@ export default class App extends PureComponent {
       return null;
     }
 
-    const attributes = TileLayer.getFeatureAttributes(info.object, info.index);
+    const attributes = this.getSelectedFeatureAttributes(info.object, info.index);
     return attributes
       ? JSON.stringify(attributes, null, 2).replace(/[\{\}']+/g, '')
       : 'loading metadata...';
   }
 
   handleClosePanel() {
-    this.setState({selectedAttribute: null});
+    this.setState({selectedFeatureAttributes: null});
+  }
+
+  getSelectedFeatureAttributes(tile, featureIndex) {
```

Comment:
Good improvement, but ultimately this should be a function in the tiles module, not the app. It does not depend on any UI or deck.gl knowledge.

Reply:
Done